### DISCUSSION TITLE
Allow spreading into custom type and using capture at the same time

### DISCUSSION THREAD

**Discussion by inoas** [gleam-lang/gleam#3106] [Ideas & suggestions]
I can write this:

```gleam
select_queries
|> combined_query_remove_order_by_from_selects()
|> CombinedQuery(
  kind: kind,
  select_queries: _,
  limit_offset: NoLimitOffset,
  order_by: [],
  epilog: NoEpilogPart,
)
```

but not this:

```gleam
sq.select
|> list.append(slct_prts)
|> SelectQuery(..sq, select: _)
```

... is there a reason for this?

**Comment by lpil** [comment]
The function capture syntax syntax only works with function calls, and the record update syntax is not a function call.

**Comment by inoas** [comment]
It is nonetheless quite unexpected, i'd expect the record update to happen first, and then the capture injected. Both work on their own but not together is what confuses me.
